server/mscalendar/availability.go

```diff
@@ -90,18 +90,56 @@ func (m *mscalendar) syncStatusUsers(userIndex store.UserIndex) (string, error)
 		return "No schedule info found", nil
 	}
 
+	m.deliverReminders(users, schedules)
 	out, err := m.setUserStatuses(users, schedules)
 	if err != nil {
-		return out, err
+		return "", err
 	}
 
 	return out, nil
 }
 
+func (m *mscalendar) deliverReminders(users []*store.User, schedules []*remote.ViewCalendarResponse) {
+	toNotify := []*store.User{}
+	for _, u := range users {
+		if u.Settings.GetReminders {
+			toNotify = append(toNotify, u)
+		}
+	}
+
+	usersByRemoteID := map[string]*store.User{}
+	for _, u := range toNotify {
+		usersByRemoteID[u.Remote.ID] = u
+	}
+
+	for _, s := range schedules {
+		user, ok := usersByRemoteID[s.RemoteUserID]
+		if !ok {
+			continue
+		}
+		if s.Error != nil {
+			m.Logger.Errorf("Error getting availability for %s: %s", user.Remote.Mail, s.Error.Message)
+			continue
+		}
+
+		mattermostUserID := usersByRemoteID[s.RemoteUserID].MattermostUserID
+		m.notifyUpcomingEvent(mattermostUserID, s.Events)
```

Comment:
0/5 since we are sending a list of events. 

```suggestion
		m.notifyUserOfUpcomingEvents(mattermostUserID, s.Events)
```

Reply:
0/5 your proposal seems too verbose. What about just `notifyUpcomingEvents`?